### PR TITLE
add tenant resolution and multi-tenant subdomain info.

### PR DESCRIPTION
**Pull request by marwen-abid**
Add the following two sections : 
* Multi-tenant considerations to **Deploy the SDP** that discusses the recommendation to acquire wildcard TLS certificates 
* Multi-tenancy section added to **Design and Architecture**

### CODE REVIEW

File: network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx [hunk 1]
```diff
@@ -37,4 +37,17 @@ There are 3 types of schemas in the database:
...
+## Multi-tenancy {#multi-tenancy}
+
+The SDP can be deployed in a multi-tenant configuration, where multiple organizations share the same instance of the SDP. Each organization is referred to as a tenant and has its own set of data and configuration. A host organization can manage multiple tenants and manage their configuration through the Admin API.
+
+### Tenant Resolution {#tenant-resolution}
+
+The SDP uses a tenant resolution strategy to determine which tenant a request belongs to. Tenant resolution is only required for unauthenticated requests, as authenticated requests include the tenant information already in the JWT token.
+
+- **Header**: the `SDP-Tenant-Name` header is used to specify the tenant name in the request. When present, this header is used to attempt resolving the tenant.
+- **Subdomain**: the SDP can use the subdomain of the request URL to resolve the tenant. For example, `tenant1.sdp.example.com` would resolve to the tenant `tenant1`.
```
Comment (marcelosalloum) [inline network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx:51]: 💡 To make it clear we're talking about the backend:
```suggestion
- **Subdomain**: the SDP can use the subdomain of the request URL to resolve the tenant. For example, `tenant1.sdp.backend.test` would resolve to the tenant `tenant1`.
```

Comment (marcelosalloum) [inline network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx:51]: I miss two more topics here:

- one is single-tenant-mode resolution (all resolves to the default tenant)
- the other topic is how the domain is also important for the frontend, where it's basically mandatory when not using single-tenant mode.

**Review by marcelosalloum — COMMENTED** [review]
Thanks for working on this! I miss two explanations, as explained below:

File: network/stellar-disbursement-platform/admin-guide/deploy-the-sdp.mdx [hunk 2]
```diff
@@ -38,6 +38,14 @@ curl -O https://raw.githubusercontent.com/stellar/stellar-disbursement-platform-
 
 You can find the full list of configurable values in the [SDP GitHub repository](https://github.com/stellar/stellar-disbursement-platform-backend/blob/develop/helmchart/sdp/README.md#stellar-disbursement-platform-sdp-parameters).
 
+There is a more detailed explanation of how to configure the SDP in the [Configuring the SDP Guide.](./configuring-sdp.mdx)
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply network/stellar-disbursement-platform/admin-guide/deploy-the-sdp.mdx]: addressed in 383e0a1e78ce90812c85776593aa9f8fd841c21b

File: network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx (hunk 1 above)
Comment (marwen-abid) [reply network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx]: addressed in 383e0a1e78ce90812c85776593aa9f8fd841c21b

File: network/stellar-disbursement-platform/admin-guide/deploy-the-sdp.mdx [hunk 3]
```diff
@@ -38,6 +38,14 @@ curl -O https://raw.githubusercontent.com/stellar/stellar-disbursement-platform-
...
+
+### Multi-tenant considerations
+
+When running the SDP in a multi-tenant configuration, you will need to acquire wildcard TLS certificates to facilitate tenant provisioning as the SDP relies on subdomains to differentiate between tenants. This will allow you to provision tenants without having to manually configure TLS certificates for each tenant. You can use a service like [Let's Encrypt](https://letsencrypt.org/) or [Namecheap](https://www.namecheap.com/security/ssl-certificates/) to acquire these certificates.
+
+For more information about multi-tenancy in the SDP, see the [Design and Architecture Guide](./design-and-architecture.mdx#multi-tenancy).
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply network/stellar-disbursement-platform/admin-guide/deploy-the-sdp.mdx]: addressed in 383e0a1e78ce90812c85776593aa9f8fd841c21b

File: network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx (hunk 1 above)
Comment (marwen-abid) [reply network/stellar-disbursement-platform/admin-guide/design-and-architecture.mdx]: addressed in 383e0a1e78ce90812c85776593aa9f8fd841c21b

**Review by marcelosalloum — APPROVED** [review]
LGTM!

**Pull request merged**